source/core/TypeAndDataManager.cpp

```diff
@@ -128,16 +128,15 @@ size_t gAllocNumWatch = 0;
 void* TypeManager::Malloc(size_t countAQ)
 {
     VIREO_ASSERT(countAQ != 0);
+    size_t allocationCount = 1;
+
+#ifdef VIREO_TRACK_MEMORY_QUANTITY
```

Comment:
I don't think we should use this directive to change the meaning of how much gets allocated only in Malloc(). There are other places that use this directive.  I would recommend adjusting only _allocationLimit in this PR and not change the meaning of what VIREO_TRACK_MEMORY_QUANTITY controls. We can discuss that later.

Reply:
Discussed with Milan. I misunderstood part of the changes. However, the way TrackAllocation and _totalAQAllocated are handled is confusing and worth revisiting later to cleanup.